linera-chain/src/block.rs

```diff
@@ -57,10 +57,16 @@ impl From<ConfirmedBlock> for HashedCertificateValue {
 impl BcsHashable for ConfirmedBlock {}
 
 impl ConfirmedBlock {
+    #[cfg(not(feature = "benchmark"))]
     pub(super) fn new(executed_block: ExecutedBlock) -> Self {
         Self { executed_block }
     }
 
+    #[cfg(feature = "benchmark")]
```

Comment:
Let's just make it public then? I think a public constructor is a lesser evil than conditional compilation.

Reply:
I don't want to allow for creating instances of `ConfirmedBlock`s "out of thin air". A correct way to do it is _only_ to promote a validated block.

Reply:
Could the benchmark code just create a validated block and promote it then?

Reply:
It could, I can go with that approach if you prefer. Didn't think of it but it feels like cheating a bit :)